File: python/p013.py
# Large sum
#
# work out the first ten digits of the sum of the following one-hundred
# 50-digit numbers.

with open("../sources/p013.txt") as f:
    source = f.read()

numbers = [int(x) for x in source.splitlines()]

result = str(sum(numbers))[:10]
print(result)

# output: 5537376230

File: go/p013.go
package main

import (
	"bufio"
	"fmt"
	"math/big"
	"os"
)

func main() {
	sum := new(big.Int)
	n := new(big.Int)
	f, _ := os.Open("../sources/p013.txt")
	scanner := bufio.NewScanner(f)

	for scanner.Scan() {
		n.SetString(scanner.Text(), 0)
		sum.Add(sum, n)
	}
	fmt.Println(sum.String()[:10])
	// output: 5537376230
}
